vault/terraform/aws/main.tf

```diff
@@ -0,0 +1,61 @@
+resource "kubernetes_namespace" "vault" {
+  metadata {
+    name = var.namespace
+    labels = {
+      "app.kubernetes.io/managed-by" = "plural"
+      "app.plural.sh/name" = "vault"
+
+    }
+  }
+}
+
+resource "aws_kms_key" "auto_unseal_key" {
+  description = "KMS key to auto-unseal Vault"
+  deletion_window_in_days = 10
+  tags = {
+    Name = "vault-auto-unseal-key"
+  }
+}
+
+resource "aws_iam_policy" "vault" {
+  name = "${var.cluster_name}-vault"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "kms:Encrypt",
+          "kms:Decrypt",
+          "kms:DescribeKey"
+        ]
+        Resource = aws_kms_key.auto_unseal_key.arn
+        Effect = "Allow"
+      }
+    ]
+  })
+}
+
+data "aws_caller_identity" "current" {
+}
+
+data "aws_eks_cluster" "plural" {
+  name = var.cluster_name
+}
+
+locals {
+  account_id = data.aws_caller_identity.current.account_id
+  oidc_provider = regex("^(https?://)(.*)$", data.aws_eks_cluster.plural.identity[0].oidc[0].issuer)[1]
+}
+
+module "assumable_role_vault" {
```

Comment:
you can actually use: https://github.com/pluralsh/module-library/tree/main/terraform/irsa-policy to encapsulate this IRSA stuff